test_generators/shuffling/core_helpers.py

```diff
@@ -0,0 +1,95 @@
+from typing import Any, List, NewType
```

Comment:
As I understand from PR description the idea of merging specs and generators is of geting rid from things like this file. Most methods from this file are in spec.

Reply:
Exactly, but we're looking to make smaller steps. The PR is already big enough. New changes to generators are upcoming though, see: #877. I like to keep generators in this PR as-is, aside from minor fixes.

Reply:
it's ok, but it's just interesting how it will look like as generated helpers code is not in repository. At least one example would be representative.

Reply:
Ok, decided to update the generators. Relative low amount of work, and it looks like this PR is a defining change right before v0.6, so we need them to be consistent. See new commits.